src/main/java/org/fenixedu/learning/servlets/FenixEduLearningContextListener.java

```diff
@@ -148,11 +150,16 @@ private static void handleThesisProposalApproval(DomainObjectEvent<Thesis> event
                     BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.subject", thesis.getStudent()
                             .getPerson().getName());
 
-            LocalizedString body =
-                    BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.body", thesis.getStudent().getPerson()
-                            .getName(), getDate(thesis.getProposedDiscussed()), String.valueOf(hasPlace(thesis)),
-                            thesis.getProposedPlace(), String.valueOf(hasTime(thesis.getProposedDiscussed())),
-                            getTime(thesis.getProposedDiscussed()), thesis.getTitle().getContent());
+            LocalizedString body = BundleUtil.getLocalizedString(Bundle.MESSAGING, "thesis.announcement.body");
+            Function<String, String> format = new Function<String, String>() {
```

Comment:
You can transform this function into a Java 8 lambda and chain the call directly. Your IDE should be able to do that for you automatically ;)